src/webapp/src/characters/charactersSlice.ts

```diff
@@ -0,0 +1,63 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+interface StateData {
+  results: StateCharacter[];
+  error?: string | null;
+}
+
+export interface StateCharacter {
+  id: number;
+  name: string;
+  species: string;
+  image: string;
+}
+interface ApiKnownError {
+  error: string;
+}
+const initialState = {
+  results: [] as StateCharacter[],
+  error: null,
+} as StateData;
+
+export const getCharacters = createAsyncThunk<
+  StateCharacter[],
+  void,
+  { rejectValue: ApiKnownError }
+>('characters/getCharacters', async (_, { rejectWithValue }) => {
+  const response = await fetch("https://rickandmortyapi.com/api/character");
+
+  if (response.status === 404) {
+    return rejectWithValue((await response.json()) as ApiKnownError);
+  }
+  const { results } = await response.json();
+
+  return results.map(({ species, image, id, name }: StateCharacter) => ({
+    species,
+    image,
+    id,
+    name,
+  }));
+});
+
+const charactersSlice = createSlice({
+  name: 'characters',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(getCharacters.pending, () => {});
+
+    builder.addCase(getCharacters.rejected, (state, action) => {
+      if (action.payload) {
+        return { ...state, error: action.payload.error };
+      }
+      return { ...state, error: action.error.message };
```

Comment:
Trying some typescript magic here. Checking if payload is there which means we will be having an error message stored on the payload. When `payload` is `undefined` the message is in the `action.error.message`. 

I think this one line might read easier. What you think?
```suggestion
        return { ...state, error: action.payload?.error || action.error?.message  };
```

Reply:
Great, thanks! 
I wanted to use the new optional chaining operator `?.` since then I read about it. But I don't have it present enough. I like the one line more, yes. More readable for sure, and cleaner! (Although I suppose in coding clean and readable are synonyms 99.99% of the cases :thinking: )

> Trying some typescript magic here. 

This magic is now doable in Javascript too, or am I wrong? It is not that you implied, I don't even know why I am asking :D